src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -2654,6 +2657,99 @@ private Process StartWithShellExecute(ProcessStartInfo startInfo)
         #endregion
     }
 
+    /// <summary>
+    /// Provides argument completion for Verb parameter.
+    /// </summary>
+    public class VerbArgumentCompleter : IArgumentCompleter
+    {
+        private const string StartProcessCommandName = "Start-Process";
+        private const string FilePathParameterName = "FilePath";
+        private const string GetCommandCommandName = "Get-Command";
+        private const string CommandParameterName = "Name";
+
+        /// <summary>
+        /// Returns completion results for verb parameter.
+        /// </summary>
+        /// <param name="commandName">The command name.</param>
+        /// <param name="parameterName">The parameter name.</param>
+        /// <param name="wordToComplete">The word to complete.</param>
+        /// <param name="commandAst">The command AST.</param>
+        /// <param name="fakeBoundParameters">The fake bound parameters.</param>
+        /// <returns>List of Completion Results.</returns>
+        public IEnumerable<CompletionResult> CompleteArgument(
+            string commandName,
+            string parameterName,
+            string wordToComplete,
+            CommandAst commandAst,
+            IDictionary fakeBoundParameters)
+        {
+            // -Verb is not supported on non-Windows platforms as well as Windows headless SKUs
+            if (!Platform.IsWindowsDesktop)
+            {
+                yield break;
+            }
+
+            // Completion: Start-Process -FilePath <path> -Verb <wordToComplete>
+            if (commandName.Equals(StartProcessCommandName, StringComparison.OrdinalIgnoreCase)
+                && fakeBoundParameters.Contains(FilePathParameterName))
+            {
+                string filePath = fakeBoundParameters[FilePathParameterName].ToString();
+
+                // Complete file verbs if extension exists
+                if (Path.HasExtension(filePath))
+                {
+                    foreach (string verb in CompleteFileVerbs(filePath, wordToComplete))
+                    {
+                        yield return new CompletionResult(verb);
+                    }
+
+                    yield break;
+                }
+
+                // Otherwise check if command is an Application to resolve executable full path with extension
+                // e.g if powershell was given, resolve to powershell.exe to get verbs
+                using (var ps = System.Management.Automation.PowerShell.Create(RunspaceMode.CurrentRunspace))
+                {
+                    var commandInfo = new CmdletInfo(GetCommandCommandName, typeof(GetCommandCommand));
+
+                    ps.AddCommand(commandInfo);
+                    ps.AddParameter(CommandParameterName, filePath);
+
+                    Collection<CommandInfo> commands = ps.Invoke<CommandInfo>();
+
+                    if (commands.Count == 1 && commands[0].CommandType == CommandTypes.Application)
```

Comment:
What is Start-Process behavior if Count > 1? Will it start first app? If so, we should check Count >= 1 here.

Reply:
@iSazonov Good point. In this case maybe simpler to just loop over commands which will not matter if 1 or more are found.

Reply:
If Start-Process pickup first app there's no point in analyzing the rest.

Reply:
I am wondering how this will work if we have multiple applications but different extensions like `test.exe` and `test.bat`. Would we have to resolve all possible verbs? If we just pick first one then we can exclude valid ones.

Reply:
I see Windows search applications based on PATHEXT. And Get-Command show me `EXE` before `BAT`. So I think it is ok to take only first element.

Reply:
Yeah looks like existing code goes through PATHEXT in order so EXE comes before BAT. I will make code changes to get first application, thanks 😊